packages/core/src/encryption.ts

```diff
@@ -47,9 +47,7 @@ const encryptData = async (iv: Uint8Array, key: CryptoKey, data: any) => {
 
   const textEncoder = new TextEncoder()
   const str = JSON.stringify(data)
-  const encoded = new Uint8Array(str.length)
```

Comment:
If we still want to use more performant `encodeInto` and not deal with UTF-8 issues, we can use `str.length * 3`: https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder/encodeInto#buffer_sizing

Reply:
I'd rather have smaller history state. Already dumping lots of stuff in there, don't want it 3x the size